audb/core/dependencies.py

```diff
@@ -375,52 +372,79 @@ def _add_media(
 
         Args:
             root: root directory
-            file: relative file path
-            archive: archive name without extension
-            checksum: checksum of file
-            version: version string
+            media: list of tuples with (
+                    relative file path,
+                    version string,
+                    archive name without extension,
+                    checksum,
+                )
 
         """
-        format = audeer.file_extension(file).lower()
+        if not media:
+            return
+
+        files = []
+        rows = []
+
+        for file, version, archive, checksum in media:
+
+            format = audeer.file_extension(file).lower()
+
+            if archive is None:
+                archive = self.archive(file)
+
+            if checksum is None:
+                checksum = self.checksum(file)
```

Comment:
Here we treat the case, where a file was removed from an archive and the archive was uploaded again without that file. In that case we have to update only the version of the remaining files in the archive. Since we only have to change the version column, I think we can also make it faster (and also cleaner) by introducing an own function.

Reply:
This I don't get directly.
When we handle removed files here, why are we recalculating checksum, bit_depth, ... for `file`?

Reply:
No, what we do is get the current values in the dependency list and then use those to update them, which obviously makes no sense. That's why I propose we add a dedicated function that only changes the version.

Reply:
OK, can you please open an issue for this.

Reply:
https://github.com/audeering/audb/issues/141